src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -262,8 +265,11 @@ private static MethodGroupResolution ResolveDelegateOrFunctionPointerMethodGroup
             }
             else
             {
-                return binder.ResolveMethodGroup(source, analyzedArguments: null, ref useSiteInfo, options: OverloadResolution.Options.IsMethodGroupConversion);
+                resolution = binder.ResolveMethodGroup(source, analyzedArguments: null, ref useSiteInfo, options: OverloadResolution.Options.IsMethodGroupConversion);
             }
+
+            Debug.Assert(!resolution.IsNonMethodExtensionMember(out _));
+            return resolution;
```

Comment:
Did you mean to remove ```return resolution;``` above? Otherwise this code can be moved to the `else` block.